docs/deployment/deploy_gardenlet.md

```diff
@@ -0,0 +1,344 @@
+            
+            **This is how it could be done for the Nginx ingress controller**
+            
+            Deploy nginx into the `kube-system` namespace in the Kubernetes cluster that should be registered as a `Seed`.
+            
+            Nginx will on most cloud providers create the service with type loadbalancer with an external ip.
+            
+            ```
+            NAME                        TYPE           CLUSTER-IP    EXTERNAL-IP
+            nginx-ingress-controller    LoadBalancer   10.0.15.46    34.200.30.30
+            ```
+            
+            Create an wildcard `A` record (e.g *.ingress.sweet-seed.<my-domain>. IN A 34.200.30.30) with your DNS provider and point it to the external ip of the ingress service. This ingress domain is later required to register the `Seed` cluster.
+            
+- **Kubeconfig for the Seed cluster**
+    
+    - Needs full admin rights
+    - Needs to be stored as a secret in the Seed cluster (configured by Gardenlet configuration `seedClientConnection.kubeconfig`)
+- **Cloud Provider credentials (e.g. AWS) for the Seed cluster**
+    
+    - Only required when the Seed should be automatically registered by the Gardenlet
+    - Secret must match the Gardenlet configuration when registering the Seed (`seedConfig.spec.secretRef`)
+    - Required permissions depend on the cloud provider. Please [see here](#TODO) for more information.
+
+## Prepare the Garden cluster
+
+1.  **Create a bootstrap token secret in the `kube-system` namespace of the garden cluster** (see [here](https://kubernetes.io/docs/reference/access-authn-authz/bootstrap-tokens/) and [here](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping/#bootstrap-tokens)).  
+    For security reasons, a bootstrap token should be used by the Gardenlet to request a signed Certificate for the Garden cluster.  
+    While it does not necessarely have to be a bootstrap token, the Client credentials need to be either `token` or `certificate` (e.g. OIDC is not supported) and it has to be possible to create a Certificate Signing Request ([CSR](https://kubernetes.io/docs/reference/access-authn-authz/certificate-signing-requests/)).
+    
+2.  **Create RBAC roles for the Gardenlet user**
+    
+    This step is only required if this is the first Gardenlet in the Gardener installation.
+    
+    The Gardenlet uses the configured Bootstrap kubeconfig in `gardenClientConnection.bootstrapKubeconfig` to request a signed certificate for the user `gardener.cloud:system:seed:<seed-name>` in the group `gardener.cloud:system:seeds`.
+    
+    Create a `ClusterRole` and `ClusterRoleBinding` that grant full admin permissions to authenticating Gardenlets.  
+    There is already an [issue open](https://github.com/gardener/gardener/issues/1724) to improve the scope of these permissions.
+    
+    Create the following resources in the Garden cluster:
```

Comment:
You could mention that this should have happened automatically if the the `controlplane` chart was used.